packages/render-fidelity-tools/src/common.ts

```diff
@@ -239,11 +242,34 @@ export class ImageComparator {
         // b, a.  here position is the index for current pixel's r , position+3
         // is index for its alpha
         const position = index * COMPONENTS_PER_PIXEL;
+        const alpha = candidateImage[position + 3];
+        if (alpha != 255) {
+          transparentPixelCount++;
+        }
 
-        if (candidateImage[position + 3] == 0) {
+        if (alpha === 0) {
           continue;
         }
 
+        let isWhitePixel = true;
+        let isBlackPixel = true;
+        for (let i = 0; i < 4; i++) {
+          const colorComponent = candidateImage[position + i];
+          if (colorComponent != 255) {
+            isWhitePixel = false;
+          }
+          if (colorComponent != 255) {
```

Comment:
Should this be `if (colorComponent != 0)`?

Reply:
Yeah it should. Thank you for finding out!